### ISSUE TITLE
论文里提到8×NVIDIA A800 (80GB) GPUs训练速度是怎么样的呀

### ISSUE THREAD

**Issue by peace-zy** [GaryGuTC/UniME-v2#6]
3%|▎         | 68/2000 [11:23:52<322:06:53, 600.21s/it]
  3%|▎         | 69/2000 [11:33:44<320:41:07, 597.86s/it]
我这边训练速度很慢，8 x A100 80G per_device_train_batch_size=256 符合预期吗？

**Comment by GaryGuTC** [comment]
Hi @peace-zy 

The current training speed is constrained by the high computational cost per batch. For example, with **m** queries each paired with **k** candidates, the total inference steps required are **m × k**. In contrast, more efficient approaches typically require only **m × 2** inference steps (commonly **k > 2**), leading to significantly lower computational overhead.

Despite this computational intensity, the method remains valuable because:
1. Using **hard negatives** often yields better results than using a larger number of easy negatives. While it demands more computation, the improvement in model effectiveness can justify the cost.

To improve training efficiency, you may consider the following recommendations:

1. **Upgrade to more powerful GPU resources** to accelerate batch processing.  
2. **Optimize the computation pipeline** by avoiding redundant inference for identical candidates within the same batch.  
3. **Adopt a more efficient training infrastructure**, such as [ms-swift](https://github.com/modelscope/ms-swift/blob/main/examples/train/embedding/train_emb.sh), which is designed for high-performance model training.

Feel free to reach out if you have further questions. We are also working to improve training efficiency in the next release.

**Comment by peace-zy** [comment]
> 1. ften yields better results tha
再请教下，k candidates是哪个参数呀torchrun $DISTRIBUTED_ARGS train.py \
  --model_name $model_name \
  --output_dir $output_dir \
  --bf16 --pooling last \
  --lora \
  --image_dir $image_dir \
  --image_resolution $image_resolution \
  --max_len $max_len \
  --logging_steps 1 \
  --lr_scheduler_type linear \
  --learning_rate $lr \
  --warmup_steps $warmup_steps \
  --save_steps $save_steps \
  --normalize True \
  --temperature 0.02 \
  --per_device_train_batch_size $per_device_train_batch_size \
  --grad_cache True \
  --gc_q_chunk_size 2 \
  --gc_p_chunk_size 2 \
  --save_safetensors True \
  --remove_unused_columns False \
  --report_to 'tensorboard' \
  --num_train_epochs 1 \
  --max_steps $max_steps \
  --run_name $run_name \
  --save_total_limit 50 \
  --save_on_each_node True \
  --resume_from_checkpoint True \
  --hard_negaitve_data_path $hard_negaitve_data_path \
  --select_hard_negative_num_4_training 8 \
  --select_hard_negative_num 8

**Comment by GaryGuTC** [comment]
Hi @peace-zy

You may want to adjust these two parameters to improve efficiency while maintaining effectiveness:

```bash
--select_hard_negative_num_4_training 8
--select_hard_negative_num 8
```

I recommend keeping their values **the same**, and not exceeding **10**.
